### PR TITLE
V1/174 175 code refactor and new design

### LINKED ISSUE #174: Code refactor
Read and refactor the code if the following those requirements:
- Repeating code: create a function to group the code
- not understandable code: add comments and refactor to be more readable
- not commented code: comment it
- unlogical code: refactor to be more selfexplainable
- not performant: refactor to be more performant
- etc.

- [x] Frontend
- [x] Backend

### LINKED ISSUE #175: Ux refactor
- [x] Create the final models
- [x] Apply the changes to the frontend

### LINKED ISSUE #177: missing delete poll route

### PR DESCRIPTION
**Pull request by fabio10000**
# To test
Go through the website and everything should work

I hope you'll enjoy the code review ;)

### CODE REVIEW

File: src/resources/js/components/AccountCreation.vue [hunk 1]
```diff
@@ -1,61 +1,100 @@
...
+        </v-row>
+        <v-row
+            align="center"
+            justify="center"
+            class="my-5"
+        >
+            <v-col
+                cols="12"
+                sm="8"
+                md="4">
```
Comment (kayoumido) [inline src/resources/js/components/AccountCreation.vue:12]: 4 is clearly too small change it to 8

File: src/resources/js/components/AccountCreation.vue [hunk 2]
```diff
@@ -1,61 +1,100 @@
...
-            :rules="fnameRules"
-            required
-            label="Prénom"
-      />
+    <v-container
+        fluid>
+        <v-row
+            align="center"
+            justify="center">
+            <span class="text-center display-4" style="color: #5BDC8E">Paul</span>
```
Comment (kayoumido) [inline src/resources/js/components/AccountCreation.vue]: Find a way to use `Baloo Bhai` as the font as it's our logo.
Here's the web font download for it https://webfonts.ffonts.net/Baloo-Bhai.font


Btw it's "Paul." not just "Paul"

File: src/resources/js/components/AccountCreation.vue [hunk 3]
```diff
@@ -1,61 +1,100 @@
 <template>
```
Comment (kayoumido) [inline src/resources/js/components/AccountCreation.vue:1]: You should add a bit of margin at the top of the page.

File: src/resources/js/components/Authentication.vue [hunk 4]
```diff
@@ -1,52 +1,95 @@
...
+            align="center"
+            justify="center"
+            class="my-5"
         >
-            Se connecter
-        </v-btn>
+            <v-col
+                cols="12"
+                sm="8"
+                md="4">
```
Comment (kayoumido) [inline src/resources/js/components/Authentication.vue:12]: 4 clearly too small, 8 works better

File: src/resources/js/components/Authentication.vue [hunk 5]
```diff
@@ -1,52 +1,95 @@
...
-            :disabled="!valid"
-            color="success"
-            class="mr-4"
-            @click="login"
+    <v-container
+        fluid>
+        <v-row
+            align="center"
+            justify="center">
+            <span class="text-center display-4" style="color: #5BDC8E">Paul</span>
```
Comment (kayoumido) [inline src/resources/js/components/Authentication.vue]: Find a way to use Baloo Bhai as the font as it's our logo.
Here's the web font download for it https://webfonts.ffonts.net/Baloo-Bhai.font

Btw it's "Paul." not just "Paul"

File: src/resources/js/components/Authentication.vue [hunk 6]
```diff
@@ -1,52 +1,95 @@
 <template>
```
Comment (kayoumido) [inline src/resources/js/components/Authentication.vue:1]: You should add a bit of margin at the top of the page.

File: src/resources/js/components/AccountEdit.vue [hunk 7]
```diff
@@ -1,124 +1,122 @@
 <template>
+    <v-container
+        fluid>
```
Comment (kayoumido) [inline src/resources/js/components/AccountEdit.vue]: just use container. it doesn't respect the model. Plus it doesn't look great

File: src/resources/js/components/AccountEdit.vue [hunk 8]
```diff
@@ -1,124 +1,122 @@
 <template>
```
Comment (kayoumido) [inline src/resources/js/components/AccountEdit.vue:1]: You forgot the password fields

File: src/resources/js/components/CreatePresentation.vue [hunk 9]
```diff
@@ -1,45 +1,74 @@
...
-    >
-      <v-text-field
-        v-model="dataForm.title"
-        :counter="20"
-        :rules="titleRules"
-        label="Titre"
-        required
-      />
+    <v-container
+        fluid>
```
Comment (kayoumido) [inline src/resources/js/components/CreatePresentation.vue]: use container. looks better

File: src/resources/js/components/EditPresentation.vue [hunk 10]
```diff
@@ -1,37 +1,66 @@
...
-        v-model="valid"
-        lazy-validation
-    >
-      <v-text-field
-            v-model="updatePresentationInfo.title"
-            :rules="titleRules"
-            label="Titre"
-      />
+    <v-container
+        fluid>
```
Comment (kayoumido) [inline src/resources/js/components/EditPresentation.vue]: use container. looks better

File: src/resources/js/components/ListPresentations.vue [hunk 11]
```diff
@@ -1,67 +1,88 @@
...
-                <v-btn icon @click.prevent="goToPresentationEdit(presentation.id)">
-                  <v-icon color="grey lighten-1">mdi-pencil-outline</v-icon>
-                </v-btn>
-              </v-list-item-action>
-
-              <v-list-item-action>
-                <v-btn icon @click.prevent="goToPresentation(presentation.id)">
-                  <v-icon color="grey lighten-1">mdi-information</v-icon>
+    <v-container
+        fluid>
```
Comment (kayoumido) [inline src/resources/js/components/ListPresentations.vue]: use container. looks better

File: src/resources/js/components/ListPresentations.vue [hunk 12]
```diff
@@ -1,67 +1,88 @@
...
+        <v-row>
+            <v-col col="10">
+                <SearchBar></SearchBar>
+            </v-col>
+        </v-row>
+        <v-divider></v-divider>
+        <v-row>
+            <v-col>
+                <v-btn
```
Comment (kayoumido) [inline src/resources/js/components/ListPresentations.vue:11]: should be on the right

File: src/resources/js/components/Presentation.vue [hunk 13]
```diff
@@ -1,50 +1,70 @@
 <template>
       <v-container
-      v-if="isLoaded"
+          fluid
```
Comment (kayoumido) [inline src/resources/js/components/Presentation.vue]: use container. looks better

File: src/resources/js/components/Presentation.vue [hunk 14]
```diff
@@ -1,50 +1,70 @@
 <template>
```
Comment (kayoumido) [inline src/resources/js/components/Presentation.vue:1]: If for some technical reason you can't follow the models with a link to create a new poll, you need to move the block to create a poll above all the polls.

File: src/resources/js/components/layout/SideBar.vue [hunk 15]
```diff
@@ -1,82 +1,36 @@
 <template>
```
Comment (kayoumido) [inline src/resources/js/components/layout/SideBar.vue:1]: It should be a bit wider. Plus according to the models, there isn't a sliding animation (which is very very annoying).

File: src/resources/js/components/Reset.vue [hunk 16]
```diff
@@ -1,25 +1,62 @@
...
-            :disabled="!valid"
-            color="success"
-            class="mr-4"
-            @click="reset"
-            >
-            Envoyer email
-      </v-btn>
-    </v-form>
+    <v-container
+        fluid>
```
Comment (kayoumido) [inline src/resources/js/components/Reset.vue]: use container. looks better

File: src/resources/js/components/ResetPassword.vue [hunk 17]
```diff
@@ -1,36 +1,65 @@
 <template>
-    <v-form
-        ref="form"
-        v-model="valid"
-        lazy-validation
-    >
+    <v-container
+        fluid>
```
Comment (kayoumido) [inline src/resources/js/components/ResetPassword.vue]: use container. looks better

File: src/resources/js/components/Reset.vue [hunk 18]
```diff
@@ -1,25 +1,62 @@
...
+        <v-row
+            align="center"
+            justify="center">
+            <span class="text-center display-4" style="color: #5BDC8E">Paul</span>
```
Comment (kayoumido) [inline src/resources/js/components/Reset.vue]: Same as for the Login and Account creation pages, wrong font

File: src/resources/js/components/Reset.vue [hunk 19]
```diff
@@ -1,25 +1,62 @@
 <template>
```
Comment (kayoumido) [inline src/resources/js/components/Reset.vue:1]: Add more spacing at the top

**Review by kayoumido — CHANGES_REQUESTED** [review]
Titles should be using the Baloo Bhai font. 
https://webfonts.ffonts.net/Baloo-Bhai.font

Is there a reason you didn't add the back arrows?

You have a problem with `conainer--fluid` it's not healty


For the `unsubscribe` you used a button and not a link!

**Review by alfavre — COMMENTED** [review]
I only tested the app, did not read the code
when the side bar is not open, the first line is `P.`
when it is open it's `Paul`
the problem is that after the first open-close, and a second after, the behavior for `P.` and `Paul` is reversed.
Another problem is the size of the blue poll choices, it's quite small (my screen is standard 16:9) the word Superman got the letter n on a second line.
when a choice is selected, the screen is weird

Screenshot from 2020-05-28 15-06-14
 )

The last weird thing is when I subscribe to a presentation, then vote on a poll, then unsubscribe from the presentation, and then search that presentation again (to subscribe again) an `Oops problème lors du chargements des résultats` message appears (but the pres. is still on screen). I suppose it happen because it tries to show the poll I answered, but can't as I'm not subscribed. When I resubscribe, it shows me the answer I gave last time I was subscribed

**Review by kayoumido — APPROVED** [review]

**Review by alfavre — APPROVED** [review]
ok now

**Pull request merged**